plugins/entando-plugin-jpdigital-exchange/src/test/java/org/entando/entando/web/digitalexchange/install/DigitalExchangeInstallResourceIntegrationTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import org.springframework.core.io.Resource;
+
+@ActiveProfiles("DEinstallTest")
+public class DigitalExchangeInstallResourceIntegrationTest extends AbstractControllerIntegrationTest {
+
+    private static final String BASE_URL = "/digitalExchange";
+
+    private static File tempZipFile;
+
+    @BeforeClass
+    public static void setupZip() {
+        tempZipFile = ComponentZipUtil.getTestWidgetZip();
+    }
+
+    @AfterClass
+    public static void deleteZip() {
+        if (tempZipFile != null) {
+            tempZipFile.delete();
+        }
+    }
+
+    @Autowired
+    private IWidgetService widgetService;
+
+    @Autowired
+    private IStorageManager storageManager;
+
+    @Autowired
+    private IInitializerManager initializerManager;
+
+    @After
+    public void cleanup() throws Exception {
+        storageManager.deleteDirectory("de_components", true);
+    }
+
+    @Configuration
+    @Profile("DEinstallTest")
+    public static class TestConfig {
+
+        @Bean
+        @Primary
+        public DigitalExchangeOAuth2RestTemplateFactory getRestTemplateFactory() {
+            return new DigitalExchangesMocker()
+                    .addDigitalExchange(DE_1_ID, (request) -> {
+                        Resource resource = mock(Resource.class);
+                        try {
+                            when(resource.getInputStream()).thenReturn(new FileInputStream(tempZipFile));
+                        } catch (IOException ex) {
+                            throw new UncheckedIOException(ex);
+                        }
+                        return resource;
+                    })
+                    .initMocks();
+        }
+    }
+
+    @Test
+    public void shouldStartInstallationJob() throws Exception {
+
+        String componentCode = "test_widget";
+
+        ResultActions result = createAuthRequest(get(BASE_URL + "/{exchange}/install/{component}", DE_1_ID, componentCode)).execute();
+
+        result.andExpect(jsonPath("$.errors").isEmpty())
+                .andExpect(jsonPath("$.metaData").isEmpty())
+                .andExpect(jsonPath("$.payload").isString());
+
+        String jsonResponse = result.andReturn().getResponse().getContentAsString();
+
+        SimpleRestResponse<String> response = new ObjectMapper()
+                .readValue(jsonResponse, new TypeReference<SimpleRestResponse<String>>() {
+                });
+
+        String jobId = response.getPayload();
+        assertThat(jobId).hasSize(20);
+
+        InstallationStatus status = InstallationStatus.CREATED;
+        int attempts = 0;
+        while (status != InstallationStatus.COMPLETED && attempts < 10) {
+            status = checkJobStatus(jobId);
+            assertThat(status).isNotEqualTo(InstallationStatus.ERROR);
+            attempts++;
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException ex) {
```

Comment:
Maybe rename the "ex" parameter to "ignored" to make it clear that the block was left empty on purpose